signer.go

```diff
@@ -86,16 +87,31 @@ func (s SharedSecretSigner) signStep(step reflect.Value) (interface{}, error) {
 		rawCommand, hasCommands = copy["commands"]
 		if !hasCommands {
 			// no commands to sign
-			return copy, nil
+			rawCommand = ""
 		}
 	}
 
+	// extract the plugin declaration for signing
+	extractedPlugins := ""
+	var err error
+	if plugins, hasPlugins := copy["plugins"]; hasPlugins {
+		extractedPlugins, err = s.extractPlugins(plugins)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// no plugins or commands -- nothing to do
+	if rawCommand == "" && extractedPlugins == "" {
```

Comment:
empty commands will actually have a signature, so might be worth letting this through (?)

Reply:
Yup, agreed.

Reply:
Will leave this check in, as it is the basis of "is this a command step"?